### ISSUE TITLE
[Bug]: Engine-resolved prefix-cache match unit is not propagated to workers

### ISSUE THREAD

**Issue by QHarshil** [vllm-project/vllm#58020]
### Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
PyTorch version              : 2.13.0+cu130
Python version               : 3.12.14 (main, Sep  1 2026, 14:16:52) [Clang 22.1.3 ] (64-bit runtime)
Python platform              : Linux-6.18.33.2-microsoft-standard-WSL2-x86_64-with-glibc2.43
Is CUDA available            : True
GPU models and configuration : GPU 0: NVIDIA GeForce RTX 4070 Laptop GPU
vLLM Version                 : 0.28.1rc1.dev1263+g382970ee6 (git sha: 382970ee6)
```

</details>

Observed on upstream main at 382970ee6ca490aeaaaf4e32c53695b581ff61ba.

### 🐛 Describe the bug

`EngineCore` resolves the prefix-cache match unit from the scheduler's view of the
KV cache groups and uses it on the scheduler side, but does not propagate the
resolved value to workers. A worker-side consumer therefore derives its own:

```python
# vllm/model_executor/layers/mamba/checkpoint.py:86
hash_block_size = self.vllm_config.cache_config.prefix_match_unit or block_size
```

`block_size` there is the worker's own `kv_cache_spec.block_size`.
`prefix_match_unit` is the user's request, and `None` means derive it from the
runtime KV cache group geometry.

For multiple groups the engine resolves `hash_block_size` as the GCD of the
prefix-cacheable group block sizes, which can be strictly finer than any one
group's block size. `resolve_kv_cache_block_sizes` already models that case:
`has_partial_mamba_group` branches on an align-mode Mamba group with
`block_size > hash_block_size`.

The geometry is one that ships in-tree. `tests/v1/core/test_mamba_align_chunk_split.py`
mirrors a Qwen3.6-27B deployment with attention block 16 and mamba block 1600, and
builds its manager with `scheduler_block_size=1600, hash_block_size=16`. With
`prefix_match_unit=None` the resolver's GCD is 16, while the worker fallback picks
its local 1600.

```python
import torch
from vllm.engine.arg_utils import EngineArgs
from vllm.v1.core.kv_cache_utils import resolve_kv_cache_block_sizes
from vllm.v1.kv_cache_interface import (
    FullAttentionSpec, KVCacheConfig, KVCacheGroupSpec, MambaSpec,
)

vllm_config = EngineArgs(
    model="Qwen/Qwen3.5-0.8B", max_model_len=4096, enable_prefix_caching=True
).create_engine_config()

kv_cache_config = KVCacheConfig(
    num_blocks=10000,
    kv_cache_tensors=[],
    kv_cache_groups=[
        KVCacheGroupSpec(["full_layer"], FullAttentionSpec(
            block_size=16, num_kv_heads=1, head_size=1, dtype=torch.float32)),
        KVCacheGroupSpec(["mamba_layer"], MambaSpec(
            block_size=1600, shapes=((1, 1),), dtypes=(torch.float32,),
            mamba_cache_mode="align")),
    ],
)

_, hash_block_size = resolve_kv_cache_block_sizes(kv_cache_config, vllm_config)
cache_config = vllm_config.cache_config
for group in kv_cache_config.kv_cache_groups:
    spec = group.kv_cache_spec
    derived = cache_config.prefix_match_unit or spec.block_size
    print(f"{type(spec).__name__:<18} block_size={spec.block_size:<5} "
          f"worker derives {derived:<5} engine resolved {hash_block_size}")
```

```text
prefix_match_unit (user request)     : None
engine resolved scheduler_block_size : 1600
engine resolved hash_block_size      : 16
FullAttentionSpec  block_size=16    worker derives 16    engine resolved 16
MambaSpec          block_size=1600  worker derives 1600  engine resolved 16
```

The Mamba checkpoint consumer computes checkpoint positions at a 1600-token unit
while the scheduler hashes and registers prefix-cache boundaries at 16-token
granularity. The checkpoint position computed at the coarser unit floors to 0, is
rejected as invalid, and the builder emits `checkpoint=None`, so the checkpoint is
dropped with no error.

The same gap applies to workers created after startup (for example, elastic EP
scale-up or restart): without an authoritative resolved value in `KVCacheConfig`,
they can only derive behavior from their local configuration rather than adopt the
geometry already chosen by the running engine.

vLLM already has this plumbing for the sibling property.
`KVCacheConfig.kv_cache_layout` is resolved by the engine core and adopted by
`Worker.initialize_from_config` through `record_kv_cache_layout`. The prefix-cache
match unit has no equivalent.

Auto-derived partial prefix matching for hybrid models landed in #46384; this
report is about the resolved unit not reaching the workers that consume it.

Expected: the unit the engine core resolves is authoritative and adopted by every
worker, and a consumer that has not been told it fails rather than deriving its own.
